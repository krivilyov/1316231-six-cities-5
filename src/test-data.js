export const testInitialState = {
  COMMON: {
    activeCity: `Amsterdam`,
    offers: [],
    reviews: [],
    sortingType: `popular`,
    mouseOverOfferId: null,
    changedBookmarkOffer: {},
    relatedOffers: [],
    bookmarkOffers: [],
  },
  USER: {
    authorizationStatus: `NO_AUTH`,
    authInfo: {},
  },
};
