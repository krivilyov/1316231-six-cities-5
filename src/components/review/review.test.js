import React from "react";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import {MemoryRouter} from "react-router-dom";
import Review from "./review";
import configureMockStore from "redux-mock-store";
import {testInitialState} from "../../test-data";

const mockStore = configureMockStore();
const store = mockStore(testInitialState);

const review = {
  comment: `comment`,
  date: `date`,
  id: 1,
  rate: 4,
  user: {
    avatarUrl: `avatar-url`,
    id: 1,
    isPro: true,
    name: `name`,
  }
};

it(`Review should render correctly`, () => {
  const tree = renderer
    .create(
        <Provider store={store}>
          <MemoryRouter>
            <Review
              review={review}
            />
          </MemoryRouter>
        </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
