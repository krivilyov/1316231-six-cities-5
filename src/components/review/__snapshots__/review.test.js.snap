// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review should render correctly 1`] = `
<li
  className="reviews__item"
>
  <div
    className="reviews__user user"
  >
    <div
      className="reviews__avatar-wrapper user__avatar-wrapper"
    >
      <img
        alt="Reviews avatar"
        className="reviews__avatar user__avatar"
        height="54"
        src="avatar-url"
        width="54"
      />
    </div>
    <span
      className="reviews__user-name"
    >
      name
    </span>
  </div>
  <div
    className="reviews__info"
  >
    <div
      className="reviews__rating rating"
    >
      <div
        className="reviews__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "80%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <p
      className="reviews__text"
    >
      comment
    </p>
    <time
      className="reviews__time"
      dateTime="date"
    >
      Invalid date
    </time>
  </div>
</li>
`;
